python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -57,6 +58,8 @@ class InvokeTermination(Exception):
 class OpenAIChatCompletionBase(OpenAIHandler, ChatCompletionClientBase):
     """OpenAI Chat completion class."""
 
+    MODEL_PROVIDER_NAME: ClassVar[str] = "openai"
```

Comment:
How would users change this if they want to? i.e. they want to use this connector for Non-OpenAI models.

Reply:
Well, this is really just a constant.  You can use whatever you want as a parameter to the decorator. Also, I'm not sure how / why you would use another **model provider** (not model) within this connector.

Reply:
one could configure an Ollama model with this connector, so the `openai` const wouldn't hold in that case.

Reply:
My mental model of the code was incorrect, then. I thought we had one connector per provider.

If connectors connect to potentially arbitrary model providers (say the OpenAI connector is used to connect to a local Ollama model, for instance), then how can we even know which provider is being user? By checking the PromptExecutionSettings?

Reply:
I agree it's not ideal. I think we can leave it like that as this is also what .Net has. Also, we shouldn't encourage people to use the OpenAI connector with Ollama now that we have the updated ollama connector.